test/validators/test_json_validators.py

```diff
@@ -0,0 +1,118 @@
+        return requests.get(url)
+
+
+def test_it_will_return_the_correct_repr():
+    v = JSONResponseValidator()
+    assert repr(v) == 'JSONResponseValidator({!r})'.format(v.schema_finder)
+
+
+def test_it_should_have_a_default_schema_finder_if_not_provided():
+    validator = JSONResponseValidator()
+    assert validator.schema_finder
+    assert type(validator.schema_finder) == PathBasedSchemaFinder
+
+
+def test_it_should_have_a_custom_schema_finder_if_provided():
+    dir_path = '/some/custom/path'
+    finder = PathBasedSchemaFinder(dir_path)
+    validator = JSONResponseValidator(schema_finder=finder)
+    assert validator.schema_finder
+    assert validator.schema_finder.schema_dir == dir_path
+
+
+def test_it_should_validate_a_valid_json_response():
+    data = {'name': {'preferred': 'some_user'}}
+    headers = {'Content-Type': 'application/vnd.elife.valid-data+json; version=1'}
+    url = 'http://localhost:1242/valid-data'
+
+    finder = PathBasedSchemaFinder('test/test_schemas')
+    validator = JSONResponseValidator(schema_finder=finder)
+    response = response_generator(url, headers, data)
+
+    assert response.headers['Content-Type']
+    assert not validator.validate(response)
+
+
+def test_it_will_validate_an_error_response():
+    data = {
+        "title": "No route found for \"GET /invalid\": Method Not Allowed (Allow: OPTIONS)",
+        "type": "about:blank"
+    }
+    headers = {'Content-Type': 'application/problem+json'}
+    url = 'http://localhost:1242/invalid'
+
+    finder = PathBasedSchemaFinder('test/test_schemas')
+    validator = JSONResponseValidator(schema_finder=finder)
+    response = response_generator(url, headers, data)
+
+    assert response.headers['Content-Type']
+    assert not validator.validate(response)
+
+
+def test_it_will_raise_error_with_no_content_type_header_in_response():
+    data = {'name': {'preferred': 'some_user'}}
+    url = 'http://localhost:1242/valid-data'
+    headers = {}
+
+    finder = PathBasedSchemaFinder('test/test_schemas')
+    validator = JSONResponseValidator(schema_finder=finder)
+    response = response_generator(url, headers, data)
+
+    with pytest.raises(MissingContentType):
+        validator.validate(response)
+
+
+def test_it_will_raise_error_with_invalid_content_type_header():
+    data = {'name': {'preferred': 'some_user'}}
+    url = 'http://localhost:1242/valid-data'
+    headers = {'Content-Type': 'some_invalid_type'}
+
+    finder = PathBasedSchemaFinder('test/test_schemas')
+    validator = JSONResponseValidator(schema_finder=finder)
+    response = response_generator(url, headers, data)
+
+    with pytest.raises(InvalidContentType):
+        validator.validate(response)
+
+
+def test_it_should_not_validate_with_invalid_json():
+    data = {'invalid': 'data'}
+    url = 'http://localhost:1242/valid-data'
+    headers = {'Content-Type': 'application/vnd.elife.valid-data+json; version=1'}
+
+    finder = PathBasedSchemaFinder('test/test_schemas')
```

Comment:
there is a bit of duplication between the different tests, so I tend to use a class with a `setUp()` method for the construction phase

Reply:
I am used to the xunit style classes as well, but am swaying to pytest's fixture paradigm, but either way you are right there was some duplication. 

I have moved the `finder` and `validator` construction logic to be fixtures now that get injected into the test functions that require them.